src/ows/src/main/java/org/geoserver/ows/util/OwsUtils.java

```diff
@@ -420,8 +420,10 @@ else if ( Set.class.isAssignableFrom( type ) ) {
     static void updateCollectionProperty(Object object, Collection newValue, Method getter)
             throws Exception {
         Collection oldValue = (Collection) getter.invoke(object, null);
-        oldValue.clear();
-        oldValue.addAll(newValue);
+        if (oldValue != null) {
+            oldValue.clear();
+            oldValue.addAll(newValue);
```

Comment:
This is called by OwsUtils.copy, which does a reflective copy based on the setters and getters of an interface. This particular code is only called in the case we have a collection with a getter but no setter. Consequently, if oldValue is null, we have no reasonable way of accessing the underlying collection (since we are dealing only with the interface).

Affected values include ResourceInfo.getMetadataLinks and ResourceInfo.getKeywords(), among others.

Fortunately, this only occurs if when we are copying into a reflectively created object (since normally these collections are initialized during a proper instantiation. Since the XStream implementation deals wiht the actual object classes instead of the interface classes, if any of these are set in the PUT request they should be correctly parsed into the final object. Otherwise, the values remain null and do not get copied into the final catalog object.

I can update the javadocs / comments to make this clearer.

Also note that previously, we would have just gotten an NPE if oldValue was null.